types/decimal_test.go

```diff
@@ -0,0 +1,298 @@
+package types
+
+import (
+	"math/big"
+	"testing"
+
+	wire "github.com/cosmos/cosmos-sdk/wire"
+	"github.com/stretchr/testify/require"
+)
+
+// create a decimal from a decimal string (ex. "1234.5678")
+func mustNewDecFromStr(t *testing.T, str string) (d Dec) {
+	d, err := NewDecFromStr(str)
+	require.NoError(t, err)
+	return d
+}
+
+//_______________________________________
+
+func TestPrecisionMultiplier(t *testing.T) {
+	res := precisionMultiplier(5)
+	exp := big.NewInt(100000)
+	require.Equal(t, 0, res.Cmp(exp), "equality was incorrect, res %v, exp %v", res, exp)
+}
+
+func TestNewDecFromStr(t *testing.T) {
+	largeBigInt, success := new(big.Int).SetString("3144605511029693144278234343371835", 10)
+	require.True(t, success)
+	tests := []struct {
```

Comment:
Include test for negative numbers as well as for malformed negative numbers
e.g. "0.-45"

Reply:
negative testcases do exist, they are added in each time. But I'll add in a malformed negative test case